src/lib/sidenav/sidenav.ts

```diff
@@ -38,6 +38,7 @@ export class MdDuplicatedSidenavError extends MdError {
   template: '<ng-content></ng-content>',
   host: {
     '(transitionend)': '_onTransitionEnd($event)',
+    '(keydown.escape)': 'handleEscapeKey()',
```

Comment:
Good to know, I saw that the dialog was using it so I went with it. I'll switch it back to use the `keyCode`.

Regarding the reasoning in the `keycodes.ts`, I don't know if this is still an issue. There's [an e2e test on the dialog](https://github.com/angular/material2/blob/master/e2e/components/dialog/dialog.e2e.ts#L18) that uses the escape key.